beacon-chain/sync/subscriber.go

```diff
@@ -91,13 +86,12 @@ func (r *Service) registerSubscribers() {
 		r.validateAttesterSlashing,
 		r.attesterSlashingSubscriber,
 	)
-	// TODO(4154): Uncomment.
-	//r.subscribeDynamic(
-	//	"/eth2/committee_index/%d_beacon_attestation",
-	//	r.currentCommitteeIndex,                     /* determineSubsLen */
-	//	noopValidator,                               /* validator */
-	//	r.committeeIndexBeaconAttestationSubscriber, /* message handler */
-	//)
+	r.subscribeDynamic(
+		"/eth2/committee_index%d_beacon_attestation",
+		r.currentCommitteeIndex,                     /* determineSubsLen */
```

Comment:
`r.currentCommitteeIndex` requires genesis state to be initialized, I think we need to block `subscribeDynamic` until `case <-stateChannel` or it'll panic at run time  👇 


```
github.com/prysmaticlabs/prysm/beacon-chain/core/helpers.RandaoMix(...)
	beacon-chain/core/helpers/randao.go:50
github.com/prysmaticlabs/prysm/beacon-chain/core/helpers.Seed(0x0, 0x0, 0xc0007ada70, 0x4, 0x8, 0x0, 0x0, 0x0, 0x0, 0x407f35c, ...)
	beacon-chain/core/helpers/randao.go:30 +0x5e
github.com/prysmaticlabs/prysm/beacon-chain/core/helpers.ActiveValidatorIndices(0x0, 0x0, 0xc00197ee10, 0x1, 0x1, 0xb, 0x0)
	beacon-chain/core/helpers/validators.go:61 +0x2aa
github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).HeadValidatorsIndices(0xc0005be0f0, 0x0, 0xc0018345b8, 0xc00183d748, 0x0, 0x0, 0x0)
	beacon-chain/blockchain/chain_info.go:119 +0xc7
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).currentCommitteeIndex(0xc0007c8c30, 0x0)
	beacon-chain/sync/subscriber_committee_index_beacon_attestation.go:21 +0x7e
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).subscribeDynamic.func1(0xc0007c8c30, 0x56ba560, 0xc001840140, 0xc0017e25a0, 0xc00182a2a0, 0xc001726040, 0x56ca7e0, 0xc000108420, 0x537527c, 0x2a, ...)
	beacon-chain/sync/subscriber.go:215 +0x121
created by github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).subscribeDynamic
	beacon-chain/sync/subscriber.go:207 +0x19f
```

Reply:
Discussed offline, HeadValidatorsIndices  should not panic, but return 0 if no state exists.

Reply:
By zero i mean empty list.